go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -556,3 +579,80 @@ func getClusterSubnetsExclusion() string {
+	aclLogging := ""
+	// ok should always be true. However, to avoid nil pointer derefence panics here, it's
+	// best to add this verification and in the worst case to resort to sane default values.
+	if nsInfo, ok := oc.namespaces[namespaceName]; ok {
+		if action == "allow" || action == "allow-related" {
+			aclLogging = nsInfo.aclLogging.Allow
+		} else if action == "drop" || action == "drop-related" {
+			aclLogging = nsInfo.aclLogging.Deny
+		}
+	} else {
+		klog.Warningf("No nsInfo object found for namespace %s", namespaceName)
+	}
+	log = aclLogging != ""
+	severity = getACLLoggingSeverity(aclLogging)
+	meter = types.OvnACLLoggingMeter
+
+	return
+}
+
+// refreshEgressFirewallLogging updates logging related configuration for all rules of this specific firewall in OVN.
+// This method can be called for example from the Namespaces Watcher methods to reload firewall rules' logging  when
+// namespace annotations change.
+// Return values are: bool - if the egressFirewall's ACL was updated or not, error in case of errors. If a namespace
+// does not contain an egress firewall ACL, then this returns false, nil instead of a NotFound error.
+func (oc *Controller) refreshEgressFirewallLogging(egressFirewallNamespace string) (bool, error) {
+	// Retrieve the egress firewall object from cache and lock it.
+	obj, loaded := oc.egressFirewalls.Load(egressFirewallNamespace)
+	if !loaded {
+		return false, nil
+	}
+
+	ef, ok := obj.(*egressFirewall)
+	if !ok {
+		return false, fmt.Errorf("refreshEgressFirewallLogging failed: type assertion to *egressFirewall"+
+			" failed for EgressFirewall of type %T in namespace %s",
+			obj, ef.namespace)
+	}
+
+	ef.Lock()
+	defer ef.Unlock()
+
+	// Find ACLs for a given egressFirewall
+	p := func(item *nbdb.ACL) bool {
+		return item.ExternalIDs["egressFirewall"] == ef.namespace
+	}
+	egressFirewallACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, p)
+	if err != nil {
+		return false, fmt.Errorf("unable to list egress firewall ACLs, err: %v", err)
+	}
+	if len(egressFirewallACLs) == 0 {
+		klog.Warningf("No egressFirewall ACLs to update in ns: %s", ef.namespace)
+		return false, nil
+	}
+
+	for i := range egressFirewallACLs {
+		// Set logging and severity
+		log, meterName, logSeverity := oc.getLogMeterSeverity(ef.namespace, egressFirewallACLs[i].Action)
+		egressFirewallACLs[i].Log = log
+		egressFirewallACLs[i].Severity = &logSeverity
+		egressFirewallACLs[i].Meter = &meterName
+	}
+	// CreateOrUpdateACLs will update all provided (non zero value) fields
+	err = libovsdbops.CreateOrUpdateACLs(oc.nbClient, egressFirewallACLs...)
+	if err != nil {
+		return false, fmt.Errorf("unable to update ACL logging in ns %s, err: %v", ef.namespace, err)
+	}
+
+	return true, nil
+}
+
+func buildEgressFwAclName(namespace string, priority int) string {
+	return fmt.Sprintf("egressFirewall_%s_%d", namespace, priority)
+}
```

Comment:
One additional nit here:
* Can the egress firewall have a name itself different than the namespace name or is it the same? If different, should we include it in the ACL name?
* Do we have a meaningful text representation for the priority? Something that conveys a useful meaning to an admin rather than a number. Outright the bat, not asking to cook something new.

Not really important since we are now synchronizing the name we can always improve later. Just commenting in case there is anything worth considering now.

Reply:
Hey thanks for the review so far.

For 1):
https://docs.openshift.com/container-platform/4.10/networking/ovn_kubernetes_network_provider/configuring-egress-firewall-ovn.html#nw-egressnetworkpolicy-object_configuring-egress-firewall-ovn
`The name for the object must be default.`
So, we only ever have a single object with name `default` per namespace and an egress fw is always identified by the namespace name.

For 2):
For EgressFW, each ACL rule actually matches a single egress firewall rule. 10000 is the starting priority, and then each subsequent rule gets the next lower prio, so 10000, 9999, 9998, 9997 ... the priorities don't have any intrinsic meaning and are not tied to the rules, rather they reflect the order in which the rules are configured and are then processed. 

So aside from the fact that the function name might be a bit confusing / off (but I also can't come up with something incredibly better other than `buildEgressFwAclRuleName` which is even longer), I think that I address your nit with answers 1) and 2)?

Wdyt? :-)

Reply:
Fair enough